vcluster/understand/volume-snapshots.mdx

```diff
@@ -0,0 +1,316 @@
+- Processes VolumeSnapshotContent resources;
+- Makes calls to the CSI driver via RPCs (CreateSnapshot, DeleteSnapshot, and, optionally, the ListSnapshots).
+
+**CSI Driver**:
+- Runs in the host or in the virtual cluster, depending on the multi-tenancy model;
+- Makes provider-specific API calls to create, delete, or list provider-specific disk snapshots.
+
+### Create volume snapshots with shared host nodes
+
+vCluster with shared nodes creates volume snapshots in the host cluster.
+
+Host cluster admin needs to complete the following setup before being able to create volume snapshots with vCluster:
+- Install common Kubernetes snapshot-controller and volume snapshot CRDs in the host cluster.
+- Installs CSI driver in the host cluster.
+- Install VolumeSnapshotClass and annotate it as default for the CSI driver.
+
+Here is a sequence diagram that shows how vCluster creates snapshots:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    box User
+    participant User as Host cluster user
+    participant CLI as vCluster CLI
+    end
+    box Host
+    participant API as K8s API
+    participant ETCD as vCluster etcd
+    participant VSCTRL as vCluster
+    participant KSCTRL as snapshot-controller
+    participant ESS as external-snapshotter sidecar
+    participant CSI as CSI Driver
+    end
+    box S3/OCI/container
+    participant SNAPSTORE as Snapshot location
+    end
+
+    User->>CLI: run vCluster snapshot create ...
+    CLI->>API: Create snapshot request Secret
+    CLI->>API: Create snapshot request ConfigMap
+
+    API-->>VSCTRL: Watch event: snapshot request ConfigMap
+    VSCTRL->>API: Create VolumeSnapshot for PVC
+
+    API-->>KSCTRL: Watch: new VolumeSnapshot
+    KSCTRL->>API: Create VolumeSnapshotContent (Pending)
+
+    API-->>ESS: Watch: new VolumeSnapshotContent (driver matches)
+    ESS->>CSI: CSI CreateSnapshot(volumeHandle, params)
+    CSI-->>ESS: snapshotHandle, size, ready
+
+    ESS->>API: Update VolumeSnapshotContent.status (ready, handle)
+    KSCTRL->>API: Update VolumeSnapshot.status (readyToUse=true, bound)
+
+    VSCTRL->>API: Update snapshot request (snapshot handle, phase=Completed)
+    VSCTRL->>SNAPSTORE: Save snapshot request details
+    VSCTRL->>ETCD: Create etcd backup
+    VSCTRL->>SNAPSTORE: Save etcd backup
+```
+
+### Create volume snapshots with private nodes
+
+vCluster with private nodes creates volume snapshots in the virtual cluster.
+
+Virtual cluster has to be created with `deploy.volumeSnapshotController.enabled` config set to `true`. With this config, vCluster installs the common snapshot-controller, and VolumeSnapshot, VolumeSnapshotContent and VolumeSnapshotClass CRDs in your virtual cluster.
+
+Then, the following setup needs to be completed to be able to create volume snapshots with vCluster:
+- Installs CSI driver in the virtual cluster.
+- Install VolumeSnapshotClass and annotate it as default for the CSI driver.
```

Comment:
Some of this overlaps with what we wrote in main doc. You might consider leaving pre-reqs all in the same page, and then on this page referencing them. If we need to add/remove steps in future, best to avoid doing it in multiple places.